docs/source/toolkit/testing.rst

```diff
@@ -0,0 +1,213 @@
+helper, use the :code:`apt_blueprint_noauth` fixture in
+:code:`globus_action_provider_tools.testing.fixtures`:
+
+.. code-block:: python
+                
+    from globus_action_provider_tools.testing.fixtures import (
+        apt_blueprint_noauth
+    )
+
+Once imported, it can be used just as any other pytest fixture. We recommend
+passing it as a parameter to another fixture which ultimately creates the app
+and returns a resource for use in tests. Provide the fixture your
+*ActionProviderBlueprint* instance under test. This will update your instance
+with stubbed out authentication. The example below shows how to create a
+:code:`client` fixture which can be used to make unauthenticated HTTP requests
+against the Action Provider:
+
+.. code-block:: python
+
+    from myapp import action_provider_blueprint
+
+    @pytest.fixture(scope="module")
+    def client(apt_blueprint_noauth):
+        apt_blueprint_noauth(action_provider_blueprint)
+        app = create_app()
+        return app.test_client()
+
+Once composed like this, you can use the :code:`client` fixture in your tests to
+receive and use a Flask *test_client* to make unauthenticated requests against
+your Action Provider:
+
+.. code-block:: python
+
+    def test_introspection_endpoint(client):
+        response = client.get("/")
+        assert response.status_code == 200
+
+
+Flask API Helpers
+-----------------
+
+If your Action Provider is built using the
+:code:`add_action_routes_to_blueprint` Flask helper, use the
+:code:`flask_helpers_noauth` fixture in
+:code:`globus_action_provider_tools.testing.fixtures*`:
+
+.. code-block:: python
+                
+    from globus_action_provider_tools.testing.fixtures import (
+        flask_helpers_noauth
+    )
+
+Once imported, simply pass the :code:`flask_helpers_noauth` fixture as a
+parameter to another fixture which creates the app and returns a resource for
+use in tests. Unlike the :code:`apt_blueprint_noauth` fixture, the
+:code:`flask_helpers_noauth` fixture does not need to be explicitly executed -
+simply passing it as a parameter is sufficient to temporarily stub out the
+Provider's authentication. An example of how to create a :code:`client` fixture
+which can be used to make unauthenticated HTTP requests against the Action
+Provider is shown below:
+
+.. code-block:: python
+
+    @pytest.fixture(scope="module")
+    def client(flask_helpers_noauth):
+        app = create_app()
+        app.config["TESTING"] = True
+        yield app.test_client()
+
+Once composed like this, you can use the :code:`client` fixture in your tests to
+receive and use a Flask *test_client* to make requests against your Action
+Provider:
+
+.. code-block:: python
+
+    def test_introspection_endpoint(client):
+        response = client.get("/")
+        assert response.status_code == 200
+
+.. note:: 
+    
+    The :code:`flask_helpers_noauth` fixture will patch the TokenChecker in a
```

Comment:
Seems less likely for others than how we do things, but an excellent note to make it clear.

Reply:
Making note of things that have bitten me before so that other's don't get hurt (: